Implement focus grabbing for relevant entries during login

marhkb opened the pull request:
Originally, focus grabbing was supposed to be part of #90.
But I decided to make a separate PR for it because pressing the Enter key,
which was supposed to solve bug #59 and was also part of #90, gave me unwanted warnings.
The two things can be separated anyway, and so #90 can be more specific to the solution of #59.

So basically this PR tries to do the following within the login flow:
1. Grabbing focus for the relevant entry after the transition from a previous page
2. Re-grabbing focus for the relevant entry after an error occurred.

marhkb committed:
Move transition to phone number page into match arm

We will be called with 'AuthorizationState::WaitPhoneNumber' anyway,
so moving it into the match arm makes it more consistent with the
other pages.
This is a preparation for a commit to implement focus grabbing for
relevant entries after the transition from a previous page. Having
everything squashed into the match arms (transition and grabbing)
will make the code clearer.

melix99 commented on src/login.rs:
```diff
@@ -203,6 +207,9 @@ impl Login {
         let self_ = imp::Login::from_instance(self);
         self_.content.set_visible_child_name("phone-number-page");
+
+        // grab focus for entry after reset
```
```suggestion
        // Grab focus for entry after reset
```

Let's keep the comment capitalization coherent. Also with the other comments below.

melix99 commented on src/login.rs:
```diff
@@ -354,13 +361,17 @@ impl Login {
                     // Otherwise just show the error in the relative label.
                     if use_empty_key {
                         self_.content.set_visible_child_name("encryption-key-page");
+                        self_.encryption_key_entry.grab_focus();
```
Both the if and else blocks have the same call, so let's remove this redundancy.

marhkb replied:
How did I miss that. Thanks.

marhkb committed:
Grab focus for entries on page changes during login

With this commit, the relevant entry for a page in the login flow is
focused after the transition from a previous page.
This behaviour will reduce the number of mouse clicks that are
neccessary for the user to log in.

marhkb committed:
Re-grab focus for entries on errors during login

With this commit, the relevant entry for a page in the login flow is
focused again after the occurrence of an error.
This behaviour has the potential to speed up the login procedure.
